js-api/src/chem.ts

```diff
@@ -429,35 +422,62 @@ export namespace chem {
       return this.inplaceSketcherDiv;
     }
 
-    static readonly FAVORITES_KEY = 'chem-molecule-favorites';
-    static readonly RECENT_KEY = 'chem-molecule-recent';
-
-    static getFavorites(): string[] {
-      return JSON.parse(localStorage.getItem(Sketcher.FAVORITES_KEY) ?? '[]');
+    async setSketcher(sketcherName: string) {
+      ui.empty(this.host);
+      ui.setUpdateIndicator(this.host, true);
+      this.changedSub?.unsubscribe();
+      const sketcherFunc = this.sketcherFunctions.find(e => e.friendlyName == sketcherName) ?? this.sketcherFunctions.find(e => e.friendlyName == DEFAULT_SKETCHER);
+      this.sketcher = await sketcherFunc!.apply();
+      this.host!.style.minWidth = '500px';
+      this.host!.style.minHeight = '400px';
+      this.host.appendChild(this.sketcher!.root);
+      await ui.tools.waitForElementInDom(this.root);
+      await this.sketcher!.init(this);
+      ui.setUpdateIndicator(this.host, false);      
+      this.changedSub = this.sketcher!.onChanged.subscribe((_: any) => {
+        this.onChanged.next(null);
+        for (let callback of this.listeners)
+          callback();
+        if (this.syncCurrentObject) {
+          const molFile = this.getMolFile();
+          if (!Sketcher.isEmptyMolfile(molFile))
+            grok.shell.o = SemanticValue.fromValueType(molFile, SEMTYPE.MOLECULE, UNITS.Molecule.MOLBLOCK);
+        }
+      });
     }
 
-    static addFavorite(molecule: string) {
-      const favorites = Sketcher.getFavorites();
-      Sketcher.checkDuplicatesAndAddToStorage(favorites, molecule, Sketcher.FAVORITES_KEY);
+    setSketcherWithExistingMolecule(){
```

Comment:
setSketcherWithExistingMolecule

Do we really need this? Why can't we simply have setSketcher and setMolecule methods?